cgp/genome.py

```diff
@@ -267,6 +267,181 @@ def randomize(self, rng: np.random.RandomState) -> None:
         # accept generated dna if it is valid
         self.dna = dna
 
+    def reorder(self, rng: np.random.RandomState) -> None:
+        """Reorder the genome
+
+        Shuffle node ordering of internal (hidden) nodes in genome without changing node behavior.
+        (Goldman 2015, DOI: 10.1109/TEVC.2014.2324539)
+
+        From a set of nodes with no dependencies (addable_nodes)
+        a (pseudo-) random node is placed at the first internal position.
+        After updating the set of addable nodes,
+        a (pseudo-) random node for the next position is picked.
+        This procedure is repeated until all nodes are placed.
+
+        Gene values are updated to correspond to the new address of the corresponding node.
+
+        Invalid gene values (from unused arities of eg. constant operator nodes)
+        are replaced by (pseudo-) random picks of permissible values for that position.
+
+        Parameters
+        ----------
+        rng : numpy.RandomState
+            Random number generator instance.
+
+        Returns
+        ----------
+        None
+        """
+        if not self._n_rows == 1:
+            Warning("n_rows must be 1, reorder is not applied")
```

Comment:
shouldn't this result in an exception? if this happens the user should fix their config, right?

Reply:
hmm, if that use case arises we can implement it. i wouldn't consider it for now